rcav2/config.py

```diff
@@ -2,80 +2,64 @@
-)
-COOKIE_FILE = os.environ.get("COOKIE_FILE", ".cookie")
-DATABASE_FILE = os.environ.get("DATABASE_FILE", ".db.sqlite3")
-JOB_DESCRIPTION_FILE = os.environ.get("JOB_DESCRIPTION_FILE")
-
-# Opik configuration
-OPIK_PROJECT_NAME = os.environ.get("OPIK_PROJECT_NAME", "rca-api")
-OPIK_URL_OVERRIDE = os.environ.get("OPIK_URL_OVERRIDE")
-
-
-def _get_opik_tags() -> list[str]:
-    """Get additional Opik tags from environment variable, defaulting to empty list.
-
-    Supports both comma-separated and space-separated tags:
-    - Comma-separated: "tag1,tag2,tag3" or "tag1, tag2, tag3"
-    - Space-separated: "tag1 tag2 tag3"
-    - Mixed: "tag1, tag2 tag3" (comma takes precedence)
-    """
-    tags_str = os.environ.get("OPIK_TAGS", "")
-    if not tags_str:
-        return []
-    # Support both comma-separated and space-separated tags
-    # If comma is present, split by comma; otherwise split by space
-    if "," in tags_str:
-        # Comma-separated: split by comma and strip each tag
-        tags = [tag.strip() for tag in tags_str.split(",")]
-    else:
-        # Space-separated: split by whitespace
-        tags = tags_str.split()
-    # Filter out empty strings
-    return [tag for tag in tags if tag]
-
-
-OPIK_TAGS = _get_opik_tags()
-
-
-# LLM configuration
-def _get_temperature() -> float:
-    """Get LLM temperature from environment variable, defaulting to 0.5."""
-    temp_str = os.environ.get("LLM_TEMPERATURE", "0.5")
-    try:
-        temperature = float(temp_str)
-        if temperature < 0.0:
-            print(
-                f"Warning: LLM_TEMPERATURE={temperature} is negative. Using default 0.5."
-            )
-            return 0.5
-        return temperature
-    except ValueError:
-        print(
-            f"Warning: LLM_TEMPERATURE={temp_str} is not a valid float. Using default 0.5."
-        )
-        return 0.5
-
-
-LLM_TEMPERATURE = _get_temperature()
+from typing import Annotated
+from pydantic import field_validator
+from pydantic_settings import BaseSettings, NoDecode
+
+
+class Settings(BaseSettings):
+    SF_DOMAIN: str
+    LLM_GEMINI_KEY: str
+
+    # Model config
+    LLM_TEMPERATURE: float = 0.5
+    RCA_IGNORE_LINES: str | None = None
+    DSPY_CACHE: bool = False
+    DSPY_DEBUG: bool = False
+
+    # Opik configuration
+    OPIK_DISABLED: bool = False
+    OPIK_PROJECT_NAME: str = "rca-api"
+    OPIK_TAGS: Annotated[list[str], NoDecode] = []
+    OPIK_URL_OVERRIDE: str | None = None
+
+    # Jira config
+    JIRA_URL: str | None = None
+    JIRA_API_KEY: str | None = None
+    # Comma-separated list of projects to search for related tickets during RCA
+    JIRA_RCA_PROJECTS: Annotated[list[str], NoDecode] = []
+
+    # Slack config
+    SLACK_API_KEY: str | None = None
+    SLACK_SEARCH_CHANNELS: Annotated[list[str], NoDecode] = []
+
+    # Internal config
+    CA_BUNDLE_PATH: str = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
```

Comment:
@adrianfusco @dasm do  we need CA_BUNDLE_PATH at all? I am guessing EOD has this setup already.